precompile/contracts/nativeminter/config.go

```diff
@@ -0,0 +1,98 @@
+// See the file LICENSE for licensing terms.
+
+package nativeminter
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/precompile/allowlist"
+	"github.com/ava-labs/subnet-evm/precompile/config"
+	"github.com/ava-labs/subnet-evm/utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/math"
+)
+
+var _ config.Config = &Config{}
+
+// Config wraps [AllowListConfig] and uses it to implement the StatefulPrecompileConfig
+// interface while adding in the ContractNativeMinter specific precompile address.
+type Config struct {
+	allowlist.Config
+	config.Uprade
+	InitialMint map[common.Address]*math.HexOrDecimal256 `json:"initialMint,omitempty"` // initial mint config to be immediately minted
+}
+
+// NewConfig returns a config for a network upgrade at [blockTimestamp] that enables
+// ContractNativeMinter with the given [admins] and [enableds] as members of the allowlist. Also mints balances according to [initialMint] when the upgrade activates.
+func NewConfig(blockTimestamp *big.Int, admins []common.Address, enableds []common.Address, initialMint map[common.Address]*math.HexOrDecimal256) *Config {
+	return &Config{
+		Config: allowlist.Config{
+			AdminAddresses:   admins,
+			EnabledAddresses: enableds,
+		},
+		Uprade:      config.Uprade{BlockTimestamp: blockTimestamp},
+		InitialMint: initialMint,
+	}
+}
+
+// NewDisableConfig returns config for a network upgrade at [blockTimestamp]
+// that disables ContractNativeMinter.
+func NewDisableConfig(blockTimestamp *big.Int) *Config {
+	return &Config{
+		Uprade: config.Uprade{
+			BlockTimestamp: blockTimestamp,
+			Disable:        true,
+		},
+	}
+}
+func (Config) Key() string { return ConfigKey }
+
+// Equal returns true if [cfg] is a [*ContractNativeMinterConfig] and it has been configured identical to [c].
+func (c *Config) Equal(cfg config.Config) bool {
+	// typecast before comparison
+	other, ok := (cfg).(*Config)
+	if !ok {
+		return false
+	}
+	eq := c.Uprade.Equal(&other.Uprade) && c.Config.Equal(&other.Config)
+	if !eq {
+		return false
+	}
+
+	if len(c.InitialMint) != len(other.InitialMint) {
+		return false
+	}
+
+	for address, amount := range c.InitialMint {
+		val, ok := other.InitialMint[address]
+		if !ok {
+			return false
+		}
+		bigIntAmount := (*big.Int)(amount)
+		bigIntVal := (*big.Int)(val)
+		if !utils.BigNumEqual(bigIntAmount, bigIntVal) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (c *Config) Verify() error {
+	if err := c.Config.Verify(); err != nil {
+		return err
+	}
+	// ensure that all of the initial mint values in the map are non-nil positive values
+	for addr, amount := range c.InitialMint {
+		if amount == nil {
+			return fmt.Errorf("initial mint cannot contain nil amount for address %s", addr)
+		}
+		bigIntAmount := (*big.Int)(amount)
+		if bigIntAmount.Sign() < 1 {
+			return fmt.Errorf("initial mint cannot contain invalid amount %v for address %s", bigIntAmount, addr)
+		}
+	}
+	return nil
```

Comment:
nit: we could verify initial mint first then return `c.Config.Verify()` (subjective) (similar with Equal)